deepchem/molnet/load_function/material_datasets.py

```diff
@@ -0,0 +1,359 @@
+            'transform_X': True
+        }
+    },
+    **kwargs) -> Tuple[List, Tuple, List]:
+  """Load band gap dataset.
+
+  Contains 4604 experimentally measured band gaps for inorganic
+  crystal structure compositions.
+  
+  Parameters
+  ----------
+  featurizer : ElementPropertyFingerprint
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  transformers : List{List of allowed transformers for this dataset}
+    A transformer that inherits from deepchem.trans.Transformer.
+  splitter : RandomSplitter
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload : bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  data_dir : str, optional
+    Path to datasets.
+  save_dir : str, optional
+    Path to featurized datasets.
+  featurizer_kwargs : dict
+    Specify parameters to featurizer, e.g. {"size": 1024}
+  splitter_kwargs : dict
+    Specify parameters to splitter, e.g. {"seed": 42}
+  transformer_kwargs : dict
+    Maps transformer names to constructor arguments, e.g.
+    {"BalancingTransformer": {"transform_x":True, "transform_y":False}}
+  **kwargs : additional optional arguments.
+
+  Returns
+  -------
+  tasks, datasets, transformers : tuple
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  References
+  ----------
+  .. [1] Zhuo, Y. et al. "Predicting the Band Gaps of Inorganic Solids by Machine Learning." J. Phys. Chem. Lett. (2018) DOI: 10.1021/acs.jpclett.8b00124.
+
+  .. [2] Dunn, A. et al. "Benchmarking Materials Property Prediction Methods: The Matbench Test Set and Automatminer Reference Algorithm." https://arxiv.org/abs/2005.00707 (2020)
+
+  Examples
+  --------
+  >> import deepchem as dc
+  >> tasks, datasets, transformers = dc.molnet.load_bandgap(reload=False)
+  >> train_dataset, val_dataset, test_dataset = datasets
+  >> n_tasks = len(tasks)
+  >> n_features = train_dataset.get_data_shape()[0]
+  >> model = dc.models.MultitaskRegressor(n_tasks, n_features)
+
+  """
+
+  # Featurize
+  logger.info("About to featurize band gap dataset.")
+  my_tasks = ['gap expt']  # machine learning targets
+
+  # Get DeepChem data directory if needed
+  if data_dir is None:
+    data_dir = DEFAULT_DIR
+  if save_dir is None:
+    save_dir = DEFAULT_DIR
+
+  # Check for str args to featurizer and splitter
+  if isinstance(featurizer, str):
+    featurizer = DEFAULT_FEATURIZERS[featurizer](**featurizer_kwargs)
+  elif issubclass(featurizer, Featurizer):
+    featurizer = featurizer(**featurizer_kwargs)
+
+  if isinstance(splitter, str):
+    splitter = DEFAULT_SPLITTERS[splitter]()
+  elif issubclass(splitter, Splitter):
+    splitter = splitter()
```

Comment:
I think these lines are for the backward compatibility and it is not  necessary for the new molnet functions to support string input

@rbharath How do you think...?

Reply:
Rather than supporting string inputs, here I could change it to just check to make sure the class type is correct and raise an error if needed.

Reply:
As a first thought, we don't quite have backwards compatibility now since we have different string arguments than the old string arguments. Perhaps it would be fine to get rid of the string arguments and just accept `Featurizer` objects?

Reply:
Sounds good, I've changed it to do a check and make sure they're an appropriate subclass, e.g. `MaterialStructureFeaturizer`